### PR TITLE
Profile Page

### PR DESCRIPTION
**Pull request by prathmeshjagtap**
Added Profile Page
Edit Profile
Edit Posts
Delete Post
Added User API calls
Get all users
Get single User
get user posts

### CODE REVIEW

File: src/frontend/features/user/userSlice.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+	allusers: [],
+	user: null,
+	userPosts: [],
+	userStatus: "idle",
+	error: "",
+};
+
+const getAllUsers = createAsyncThunk("/user/getAllUsers", async () => {
+	const { data } = await axios.get("/api/users");
+	return data;
```
Comment (ajay-mandaviya) [inline src/frontend/features/user/userSlice.js:14]: it's good if you use try catch here in side the createAsyncThunk and user and inside the catch 
return thunkAPI.rejectWithValue(error);

File: src/frontend/screens/profile/EditProfile.jsx [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+import { useDispatch, useSelector } from "react-redux";
+
+function EditProfile({ close, user }) {
+	const [profile, setProfile] = useState(user);
+	const dispatch = useDispatch();
+	const { token } = useSelector((state) => state.auth);
+	const handleChange = (e) => {
+		setProfile({
+			...profile,
+			[e.target.name]: e.target.value,
```
Comment (ajay-mandaviya) [inline src/frontend/screens/profile/EditProfile.jsx]: you can use like 
const {name , value } = e.target
[name] : value

**Review by ajay-mandaviya — APPROVED** [review]

**Review by rehanmohiuddin — APPROVED** [review]

File: src/frontend/features/user/userSlice.js (hunk 1 above)
Comment (prathmeshjagtap) [reply src/frontend/features/user/userSlice.js:14]: I don't need to catch the error message here so I am not using try catch here

File: src/frontend/screens/profile/EditProfile.jsx (hunk 2 above)
Comment (prathmeshjagtap) [reply src/frontend/screens/profile/EditProfile.jsx]: Added this changes

**Pull request merged**